### PR TITLE
`azurerm_eventhub_namespace`: support latest released properties:`local_authentication_enabled`,`public_network_access_enabled`,`minimum_tls_version`

### PR DESCRIPTION
**Pull request by xiaxyi**
1. update eventhub namespace SDK from 2021-01-01-preview to stable 2021-11-01 
2. Add SAS Auth switch for eventhub namespace

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
Once minor schema change and this is good to merge 🌵

**Comment by xiaxyi** [comment]
Thanks @katbyte updated.

**Comment by xiaxyi** [comment]
@katbyte There are two more properties released in the latest api, I added them in this pr.

**Comment by xiaxyi** [comment]
related issue: #17483

File: internal/services/eventhub/eventhub_namespace_resource.go [hunk 1]
```diff
@@ -471,6 +512,16 @@ func resourceEventHubNamespaceRead(d *pluginsdk.ResourceData, meta interface{})
...
+			if props.DisableLocalAuth != nil {
+				localAuthEnable := *props.DisableLocalAuth
+				d.Set("local_authentication_enabled", !localAuthEnable)
+			}
+			if props.PublicNetworkAccess != nil {
+				d.Set("public_network_access_enabled", *props.PublicNetworkAccess == namespaces.PublicNetworkAccessEnabled)
+			}
+			if props.MinimumTlsVersion != nil {
+				d.Set("minimum_tls_version", *props.MinimumTlsVersion)
+			}
```
Comment (tombuildsstuff) [inline internal/services/eventhub/eventhub_namespace_resource.go:524]: all 3 of these fields need to be set all of the time, with a default value if they're not returned e.g.

```
publicNetworkAccessEnabled := true
if props.PublicNetworkAccess != nil {
   *props.PublicNetworkAccess == namespaces.PublicNetworkAccessEnabled
}
d.Set("public_network_access_enabled", publicNetworkAccessEnabled)
```

File: internal/services/eventhub/eventhub_namespace_resource.go [hunk 2]
```diff
@@ -190,6 +190,29 @@ func resourceEventHubNamespace() *pluginsdk.Resource {
 				},
 			},
 
+			"local_authentication_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
```
Comment (tombuildsstuff) [inline internal/services/eventhub/eventhub_namespace_resource.go:196]: is this also the default state the API returns when provisioning on an older version and upgrading to a newer api version?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Left a couple of comments inline, we need to ensure there's always a value set for these fields into the state - meaning that we need to ensure we consider what the default value should be when this isn't returned - but if we can fix that up then this should otherwise be good to go 👍

File: internal/services/eventhub/eventhub_namespace_resource.go (hunk 2 above)
Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_resource.go:196]: yes, I tried the older version and the default value is the same

**Comment by xiaxyi** [comment]
@tombuildsstuff  Thanks for the comment, I updated the code and added the default value to set back. Feel free to let me know if it looks cool to you.

**Comment by sebader** [comment]
I think we need to get some traction on this. The latest version 3.16 already updated the API version. This now seems to have broken the event hub namespace deployment:

This used to work but does no longer:

```
resource "azurerm_eventhub_namespace" "stamp" {

  network_rulesets {
    default_action = "Deny" # Access is only allowed through the private endpoints
  }
```

```
 Error: setting network ruleset properties for Namespace (Subscription: "afffa704-8cbd-46f2-a146-b473f632ecb5"
│ Resource Group Name: "ace2ef6b5-stamp-swedencentral-rg"
│ Namespace Name: "ace2ef6b5-swedencen-evhns"): networkrulesets.NetworkRuleSetsClient#NamespacesCreateOrUpdateNetworkRuleSet: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="This update of networkRuleSets, sets zero IPRule and VirtualNetworkRules, and the DefaultAction is set to Deny. This would render the namespace resources inaccessible for data-plane requests. If the intention was to restrict data-plane requests to private links only, please set the PublicNetworkAccess property to true. Else either set the DefaultAction property to Allow OR add some allowed IPRules or VirtualNetworkRules in the update request. CorrelationId: 64d0d212-c2e9-e1e3-4df4-28dffdd64271"
│ 

```

And if I understand this PR correctly, only once it is merged can `PublicNetworkAccess` be set.

**Comment by magodo** [comment]
Tests are passing, except the following 5, which are also failing on the main branch. An investigation on these failures are in progress:

image

image

**Comment by sebader** [comment]
thanks @xiaxyi I can confirm this works again now. I replaced
```
network_rulesets {
    default_action = "Deny"
  }
```
with 

     public_network_access_enabled = false

**Pull request merged**